include/revng/Pipeline/Loader.h

```diff
@@ -197,3 +205,14 @@ struct llvm::yaml::MappingTraits<pipeline::PipelineDeclaration>
     io.mapRequired("Steps", info.Steps);
   }
 };
+
+using OptArtifactsDeclaration = std::optional<pipeline::ArtifactsDeclaration>;
+template<>
+struct llvm::yaml::MappingTraits<OptArtifactsDeclaration> {
+  static void mapping(IO &io, OptArtifactsDeclaration &info) {
+    std::string Container, Kind;
+    io.mapRequired("Container", Container);
+    io.mapRequired("Kind", Kind);
+    info = { Container, Kind };
```

Comment:
Not sure what this line is doing. I am curious more than anything else

Reply:
Basically info at the beginning is `nullopt`, so assigning struct fields to it is UB (and from what I saw while testing it does not save the fields).
So the fields are read separately and then the entire nullopt is substituted by assignment with the proper struct.

Reply:
ah, but is mapRequired evaluated on the fly? i guess so because otherwise you would have noticed this being broken.

I assumed that it build the mapping internally and then later assigned the values to all mappings.

Maybe 
```
info = {"", ""};
io.mapRequired("Container", Info->Container);
io.mapReqiired("Kind", Info->Kind); 
```
would look less spooky

Reply:
Gotcha, fixed (together with all the issues below).